core/naive-strategy.go

```diff
@@ -294,18 +297,17 @@ func (st NaiveStrategy) UnrelayedAcknowledgements(src, dst *ProvableChain, sh Sy
 }
 
 // TODO add packet-timeout support
-func relayPackets(chain *ProvableChain, seqs []uint64, sh SyncHeadersI, sender sdk.AccAddress) ([]sdk.Msg, error) {
+func relayPackets(ctx QueryContext, chain *ProvableChain, seqs []uint64, sender sdk.AccAddress) ([]sdk.Msg, error) {
```

Comment:
The names `relayPackets` and `relayAcks` seem not intuitive.
How about using different names, such as `collectPackets` and `collectAcks`?